### ISSUE TITLE
Trainning my own dataset  

### ISSUE THREAD

**Issue by casiahnu** [Tramac/awesome-semantic-segmentation-pytorch#126]
Each image in my dataset is 256*256, and is divided into two categories: background and subject. 
It contains 
-train
--image
--mask
-test
--image
--mask

Could you please tell me which dataloader.py should I refer and modify? Thank You!

**Comment by zuo-ux** [comment]
I have the same problem.

**Comment by lichen14** [comment]
hi, did you solve it? I have the same problem. @casiahnu @zuo-ux

**Comment by zuo-ux** [comment]
> hi, did you solve it? I have the same problem. @casiahnu @zuo-ux

no

**Comment by Tramac** [comment]
It is best to modify on `cityscapes.py`.

**Comment by zuo-ux** [comment]
> It is best to modify on `cityscapes.py`.

how to operate dataloader？

**Comment by casiahnu** [comment]
> It is best to modify on `cityscapes.py`. Do you have WeChat？I have not had enough time to follow github recently, maybe solve the problem by WeChat.

Thank you. My wechat is 18810220665. What's the function of segbase.py?  Look forward to your reply.

**Comment by zuo-ux** [comment]
> It is best to modify on `cityscapes.py`. Do you have WeChat？I have not had enough time to follow github recently, maybe solve the problem by WeChat.

Thank you!My wechat is 18761601876,Look forward to your reply.

**Comment by Qy19970630** [comment]
@casiahnu  Hi,did you solve the problem?

**Comment by Qy19970630** [comment]
@zuo-ux Hi,did you solve the problem?

**Comment by Zhang-O** [comment]
> > It is best to modify on `cityscapes.py`. Do you have WeChat？I have not had enough time to follow github recently, maybe solve the problem by WeChat.
> 
> Thank you!My wechat is 18761601876,Look forward to your reply.

Have you added wechat and solved your problem??

**Comment by Kittywyk** [comment]
> Each image in my dataset is 256*256, and is divided into two categories: background and subject.
> It contains
> -train
> --image
> --mask
> -test
> --image
> --mask
> 
> Could you please tell me which dataloader.py should I refer and modify? Thank You!

Hi, I met the same problem.
I've already written a dataloader file and renamed it to mydata.py for example. I also added the name "mydata" in code as an argument. 
But when I ran the command " python train.py --model fcn32s --backbone vgg16 --dataset mydata --lr 0.01 --epochs 50" ,the error occured: train.py: error: argument --dataset: invalid choice: 'mydata' (choose from 'pascal_voc', 'pascal_aug', 'ade20k', 'citys', 'sbu')
Do u know what should I modify then? 
Thanks for ur help!
